### PR TITLE
Replace Textarea component with TextareaWithCharCount component

### PR DESCRIPTION
**Pull request by juliatan**
# Description

- Extract out `Textarea` with character count to a new component called `TextareaWithCharCount`
- Replace existing `Textarea` components throughout app with new component
- Ticket: https://www.notion.so/joinzoe/Refactor-TextArea-Component-with-Character-Limit-9ae797d55e494daba21787c5aea9c542

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevant
- [x] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

Should be part of Formik refactor as there's probably an opportunity to standardise how this looks throughout the app.

### CODE REVIEW

**Review by aikewoody — DISMISSED** [review]
I love this PR!

File: src/components/TextareaWithCharCount.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  placeholder?: string;
+  rowSpan?: number;
+  style?: StyleProp<TextStyle>;
+  value: string | null;
+}
+
+export default function TextareaWithCharCount(props: ITextareaWithCharCountProps) {
+  const [charCount, setCharCount] = useState(0);
+
+  const maxLength = props.maxLength || 1000;
```
Comment (bionara) [inline src/components/TextareaWithCharCount.tsx:20]: May be good to set 1000 as a const, but happy as is.

**Review by bionara — APPROVED** [review]
very niiiiiiiice 👌

File: src/components/TextareaWithCharCount.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <>
+      <Textarea
+        bordered={props.bordered || false}
+        maxLength={maxLength}
+        onChangeText={(value) => {
+          props.onChangeText(value);
+          setCharCount(value.length);
+        }}
+        placeholder={props.placeholder}
+        rowSpan={props.rowSpan || 5}
```
Comment (bionara) [inline src/components/TextareaWithCharCount.tsx:32]: if making the above a const, do so also for the 5 here pls :)

File: src/components/TextareaWithCharCount.tsx (hunk 1 above)
Comment (juliatan) [reply src/components/TextareaWithCharCount.tsx:20]: Yep, good shout - thanks! Will change for this and row span.

**Review by selvach — APPROVED** [review]

**Pull request merged**